packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -427,6 +432,19 @@ class DevFS {
     });
     if (dirtyEntries.isNotEmpty) {
       printTrace('Updating files');
+      if (generator != null) {
+        dirtyEntries.forEach((Uri deviceUri, DevFSContent content) {
+          if (content is DevFSFileContent) {
+            generator.invalidate(content.file.uri);
+          }
+        });
+        final DeltaProgram delta = await generator.computeDelta();
+        final Program program = delta.newProgram;
+        writeProgramToBinary(program, mainPath + ".dill");
```

Comment:
Note: I don't believe it is necessary to use `serializeProgram` here because `newProgram` is already truncated to contain just the libraries that need to be serialized. Nevertheless it will be good to validate that this works as expected.